openff/fragmenter/fragment.py

```diff
@@ -290,43 +290,70 @@ def _find_functional_groups(
+    ) -> List[BondTuple]:
         """Finds the rotatable bonds in a molecule *including* rotatable double
         bonds.
 
-        Notes
-        -----
-        * This does not find terminal rotatable bonds such as -OH, -NH2 -CH3.
-
-        Todos
-        -----
-        * Add the option to build fragments around terminal torsions (-OH, -NH2, -CH3)
-
         Parameters
         ----------
         molecule
             The molecule to search for rotatable bonds.
+        rotatable_bond_smarts
+            An optional list of SMARTS patterns that should be used to identify the bonds
+            within the parent molecule to grow fragments around. Each SMARTS pattern
+            should include **two** indexed atoms that correspond to the two atoms
+            involved in the central bond.
+
+            If no pattern is provided fragments will be constructed around all 'rotatable
+            bonds'. A 'rotatable bond' here means any bond matched by a
+            `[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]` SMARTS pattern with the added
+            constraint that the **heavy** degree (i.e. the degree excluding hydrogen) of
+            both atoms in the bond must be >= 2.
 
         Returns
         -------
-            A list of the rotatable map indices of the atoms which form the rotatable
+            A list of the **map** indices of the atoms that form the rotatable
             bonds, ``[(m1, m2),...]``.
         """
 
-        matches = molecule.chemical_environment_matches(
-            "[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]"
-        )
+        if rotatable_bond_smarts is None:
+
+            matches = molecule.chemical_environment_matches(
+                "[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]"
+            )
+
+        else:
+
+            matches = [
+                match
+                for smarts in rotatable_bond_smarts
+                for match in molecule.chemical_environment_matches(smarts)
+            ]
+
+            if not all(len(match) == 2 for match in matches):
+
+                raise ValueError(
+                    f"The `rotatable_bond_smarts` pattern ({rotatable_bond_smarts}) must define exactly "
+                    f"two indexed atoms to match."
+                )
+
         unique_matches = {tuple(sorted(match)) for match in matches}
 
-        # Drop bonds without a heavy degree of at least 2 on each end to avoid finding
-        # terminal bonds
-        def heavy_degree(atom_index: int) -> int:
-            atom = molecule.atoms[atom_index]
-            return sum(1 for atom in atom.bonded_atoms if atom.atomic_number != 1)
+        if rotatable_bond_smarts is None:
 
-        unique_matches = {
-            match for match in unique_matches if all(heavy_degree(i) > 1 for i in match)
-        }
+            # Drop bonds without a heavy degree of at least 2 on each end to avoid
+            # finding terminal bonds
+            def heavy_degree(atom_index: int) -> int:
```

Comment:
Do we want to expose this as a class setting to allow fragments around terminal groups? This could cause confusion if the user passes a pattern that allows them but fragmenter is hardcoded to ignore.

Reply:
I think the dropping of terminal bonds only happens if no custom patterns are provided. In principle I think the user could pass:

```target_bond_smarts='[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]'```

and they'd be retained or

```target_bond_smarts='[!#1][!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2][!#1]'```

and they'd be dropped like the default behaviour